elements/rh-audio-player/test/rh-audio-player.spec.ts

```diff
@@ -0,0 +1,524 @@
+      });
+
+      checkButtonsInMode('full');
+
+      describe('on a mobile display', function() {
+        beforeEach(async function() {
+          await setViewport({ width: 400, height: 800 });
+          await element.updateComplete;
+          await aTimeout(100);
+        });
+
+        checkButtonsInMode('full-mobile');
+
+        it('poster is not visible', function() {
+          expect(getShadowElementBySelector('#poster')).to.not.be.visible;
+        });
+      });
+
+      describe('pressing play button', function() {
+        beforeEach(async function() {
+          getShadowElementBySelector('#full-play')?.click();
+          await element.updateComplete;
+        });
+
+        it('plays', function() {
+          expect(element.paused, 'state').to.be.false;
+          expect(element.icon, 'icon').to.equal(RhAudioPlayer.icons.pause);
+        });
+
+        describe('then pressing pause button', function() {
+          beforeEach(async function() {
+            getShadowElementBySelector('#full-play')?.click();
+            await element.updateComplete;
+          });
+
+          it('pauses', function() {
+            expect(element.paused, 'state').to.be.true;
+            expect(element.icon, 'icon').to.equal(RhAudioPlayer.icons.play);
+          });
+        });
+      });
+    });
+
+    it('disables rewind', function() {
+      expect(getShadowElementBySelector('#rewind').disabled).to.be.true;
+    });
+
+    describe('clicking the forward button', function() {
+      let starttime: number;
+      before(async function() {
+        starttime = element.currentTime;
+        getShadowElementBySelector('#forward')?.click();
+        await element.updateComplete;
+      });
+      it('skips forward', function() {
+        expect(element.currentTime > starttime).to.be.true;
+        expect(element.currentTime).to.equal(starttime + 15);
+      });
+    });
+
+    describe('clicking the forward button', function() {
+      let starttime: number;
+      before(async function() {
+        starttime = element.currentTime;
+        getShadowElementBySelector('#rewind')?.click();
+        await element.updateComplete;
+      });
+      it('skips forward', function() {
+        expect(element.currentTime < starttime).to.be.true;
+        expect(element.currentTime).to.equal(starttime - 15);
+      });
+    });
+
+    describe('setting the volume property to 0', function() {
+      let snapshot: A11ySnapshot;
+      beforeEach(async function() {
+        element.volume = 0;
+        await element.updateComplete;
+        snapshot = await a11ySnapshot();
+      });
+      it('sets muted', function() {
+        expect(element.muted).to.be.true;
+      });
+      // TODO: check for 'mute'/'unmute' aria label on button
+      // TODO: do this assertion only in full mode
+      it.skip('sets volume slider to 0', function() {
+        expect(snapshot?.children?.find(x => x.role === 'slider')?.value).to.equal(0);
```

Comment:
leaving this here for now. maybe if we roll our own we can get access to the element in the snapshot, which would let us write actions which query the a11y tree instead of the shadow root

https://github.com/microsoft/playwright/issues/16159#issuecomment-1209743311